File: Second Year/Semester II/Web Programming/lab8_aspnet/lab8_aspnet/Models/Reservation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace lab8_aspnet.Models
{
    [Table("reservations")]
    public class Reservation
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [Column("room_id")]
        public int RoomId { get; set; }

        [Required]
        [Column("user_name")]
        public string UserName { get; set; } = string.Empty;

        [Required]
        [Column("start_date")]
        public DateOnly StartDate { get; set; }

        [Required]
        [Column("end_date")]
        public DateOnly EndDate { get; set; }

        [ForeignKey("RoomId")]
        public Room? Room { get; set; }
    }
}

File: Second Year/Semester II/Web Programming/lab9_node/src/models/Reservation.js
class Reservation {
    constructor({ id, room_id, room_number, hotel_name, user_name, start_date, end_date }) {
        this.id = id;
        this.roomId = room_id;
        this.roomNumber = room_number;
        this.hotelName = hotel_name;
        this.userName = user_name;
        this.startDate = start_date;
        this.endDate = end_date;
    }
}

module.exports = Reservation;